sql/catalyst/src/test/scala/org/apache/spark/sql/types/DecimalSuite.scala

```diff
@@ -299,4 +299,19 @@ class DecimalSuite extends SparkFunSuite with PrivateMethodTester with SQLHelper
       assert(Decimal.fromStringANSI(UTF8String.fromString(string)) === Decimal(string))
     }
   }
+
+  test("SPARK-37451: Performance improvement regressed String to Decimal cast") {
```

Comment:
According to the above description, this causes a different result accidentally?
> The result is null since SPARK-32706.